rpcs3/Emu/NP/np_structs_extra.cpp

```diff
@@ -438,4 +438,121 @@ namespace extra_nps
 		sceNp.warning("recordDate: %d", data->recordDate.tick);
 	}
 
+	void print_SceNpMatchingAttr(const SceNpMatchingAttr* data)
+	{
+		sceNp.warning("SceNpMatchingAttr:");
+		sceNp.warning("next: 0x%x", data->next);
+		sceNp.warning("type: %d", data->type);
+		sceNp.warning("id: %d", data->id);
+
+		if (data->type == SCE_NP_MATCHING_ATTR_TYPE_BASIC_BIN || data->type == SCE_NP_MATCHING_ATTR_TYPE_GAME_BIN)
+		{
+			sceNp.warning("ptr: *0x%x", data->value.data.ptr);
+			sceNp.warning("size: %d", data->value.data.size);
+			sceNp.warning("data:\n%s", fmt::buf_to_hexstring(static_cast<u8 *>(data->value.data.ptr.get_ptr()), data->value.data.size));
```

Comment:
what if data->value.data.ptr is not valid ?

Reply:
It would crash the PS3 as it will read memory from that address to send it to the server.